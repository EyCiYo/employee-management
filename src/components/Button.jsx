import "../styles/Button.css";

const Button = (prop) => {
    const { label } = prop;
    return (
        <button
            className="login-btn"
            type={prop.type}
            style={prop.style}
            onClick={prop.handleClick}
        >
            {label}
        </button>
    );
};

export default Button;
