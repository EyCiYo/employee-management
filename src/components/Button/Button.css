:root{
    --emphasis-clr: rgb(1, 0, 34);
}

.login-btn{
    padding:1rem;
    background: #03AEEE;
    color:white;
    border:1px transparent;
    border-radius:5px;
    width:100%;
    cursor:pointer;
    font-size:18px;
    transition: all 0.3s ease-in-out;
    outline:1.5px transparent;
}
.login-btn:hover{
    outline: 1.5px solid var(--emphasis-clr);
    color: var(--emphasis-clr)
}