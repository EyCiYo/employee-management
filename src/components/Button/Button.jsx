import "./Button.css";
import React from "react";

const Button = (prop) => {
    const { label } = prop;
    return (
        <button
            data-testid="button-test-id"
            className="login-btn"
            type={prop.type}
            style={prop.style}
            onClick={prop.handleClick}
        >
            {label}
        </button>
    );
};

export default Button;
