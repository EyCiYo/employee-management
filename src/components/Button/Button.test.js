/* eslint-disable no-undef */
import { fireEvent, render } from "@testing-library/react";
import Button from "./Button";
import React from "react";

describe("Button test cases", () => {
    test("if button rendered succesfully", () => {
        const { getByTestId } = render(<Button />);
        expect(getByTestId("button-test-id")).toBeTruthy;
    });

    test("if onClick is triggered once", () => {
        const handleClick = jest.fn();
        const { getByTestId } = render(<Button handleClick={handleClick} />);
        fireEvent.click(getByTestId("button-test-id"));
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
    test("if onClick is triggered not more than once", () => {
        const handleClick = jest.fn();
        const { getByTestId } = render(<Button handleClick={handleClick} />);
        fireEvent.click(getByTestId("button-test-id"));
        expect(handleClick).not.toHaveBeenCalledTimes(2);
    });
    test("if snapshot is matching", () => {
        const handleClick = jest.fn();
        const { baseElement } = render(
            <Button
                type="button"
                label="button"
                style={{ color: "white" }}
                handleClick={handleClick}
            />
        );
        expect(baseElement).toMatchSnapshot();
    });
});
