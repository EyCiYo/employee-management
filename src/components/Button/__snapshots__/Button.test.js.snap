// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases if snapshot is matching 1`] = `
<body>
  <div>
    <button
      class="login-btn"
      data-testid="button-test-id"
      style="color: white;"
      type="button"
    >
      button
    </button>
  </div>
</body>
`;
